src/wallet/rpcdump.cpp

```diff
@@ -151,8 +154,23 @@ UniValue importprivkey(const JSONRPCRequest& request)
         // whenever a key is imported, we need to scan the whole chain
         pwalletMain->UpdateTimeFirstKey(1);
```

Comment:
Is this still valid, also if we do a partial rescan?

Reply:
Maybe not. The only place this gets used is in `[ScanForWalletTransactions](https://github.com/dogecoin/dogecoin/blob/1.14.7-dev/src/wallet/wallet.cpp#L1531-L1534)`, so we could fake up an intelligent value to avoid doing any work in that section... but all that code does is pick an efficient starting height, if one's not already provided.

A simple approach is to move this code to a conditional where we lack a `height`, so we preserve the existing behavior and don't have to think too hard about setting an appropriate value otherwise.

Reply:
+1 for the simple approach 😁